modin/core/io/file_dispatcher.py

```diff
@@ -258,16 +257,19 @@ def file_exists(cls, file_path, storage_options=None):
             Whether file exists or not.
         """
         if isinstance(file_path, str):
-            match = S3_ADDRESS_REGEX.search(file_path)
-            if match is not None:
-                if file_path[0] == "S":
-                    file_path = "{}{}".format("s", file_path[1:])
-                S3FS = import_optional_dependency(
-                    "s3fs", "Module s3fs is required to read S3FS files."
-                )
+            if fsspec.core.split_protocol(file_path)[0] in (
+                "s3",
+                "S3",
+                "https",
+                "http",
+            ):
+                # `file_path` may start with a capital letter, which isn't supported by `fsspec.core.url_to_fs` used below.
+                file_path = file_path[0].lower() + file_path[1:]
```

Comment:
this looks like an `fsspec` bug

Reply:
`fsspec` answered in this issue https://github.com/fsspec/filesystem_spec/issues/1009